core/trino-main/src/main/java/io/trino/server/remotetask/HttpRemoteTask.java

```diff
@@ -598,8 +598,8 @@ private synchronized void processTaskUpdate(TaskInfo newValue, List<SplitAssignm
             }
         }
 
-        // increment pendingRequestsCounter by 1 when there are still pending splits
-        if (pendingSourceSplitCount > 0) {
+        if (adaptiveUpdateRequestSizeEnabled && pendingSourceSplitCount > 0) {
```

Comment:
I'm not sure that adding a boolean check in front of an integer comparison is more efficient, especially not enough so to reduce lock-held time.

Reply:
This will be extracted as separate PR.